feat: Add LanceDB vector store support and configuration options

NaccOll opened the pull request:
## Context

<!-- Brief description of WHAT you’re doing and WHY. -->

Add LanceDB as an option for code-index vector storage to provide an out-of-the-box experience

## Implementation

<!--

Some description of HOW you achieved it. Perhaps give a high level description of the program flow. Did you need to refactor something? What tradeoffs did you take? Are there things in here which you’d particularly like people to pay close attention to?

-->

By default, qdrant will continue to be used for vector storage, which will not impact the experience for existing users.

LanceDB has been added as a new option to provide out-of-the-box support for users who prefer not to enable qdrant. This allows users to customize the vector storage location and ensures that different vector storage provider produce the same results when using the same embed model.

This pull request is currently in draft form and is being used to discuss the feasibility of this solution with the community and conduct performance testing.

## Screenshots

<img width="1553" height="1420" alt="image" src="https://github.com/user-attachments/assets/2886d030-e90c-4452-bdd2-6636641098bb" />


## How to Test

<!--

A straightforward scenario of how to test your changes will help reviewers that are not familiar with the part of the code that you are changing but want to see it in action. This section can include a description or step-by-step instructions of how to get to the state of v2 that your change affects.

A "How To Test" section can look something like this:

- Sign in with a user with tracks
- Activate `show_awesome_cat_gifs` feature (add `?feature.show_awesome_cat_gifs=1` to your URL)
- You should see a GIF with cats dancing

-->

1. Select "LanceDB" as Vector Storage Provider
2. Start Indexing
3. Use codebase_search
4. change file and trigger re-index
5. Restart vscode and test re-scan

## Get in Touch

<!-- We'd love to have a way to chat with you about your changes if necessary. If you're in the [Kilo Code Discord](https://kilocode.ai/discord), please share your handle here. -->

NaccOll

chrarnoldus commented:
@NaccOll thanks for bringing this PR over. We would like to test it a bit, but I think we can accept it.

NaccOll committed:
feat: Add LanceDB vector store support and configuration options

- Implemented synchronous storage path retrieval for LanceDB vector store.
- Updated CodeIndexPopover component to include vector store provider selection (LanceDB or Qdrant).
- Added settings for LanceDB vector store directory in the ExtensionStateContext.
- Enhanced localization files to support new settings related to vector store provider and LanceDB directory.
- Updated UI to conditionally render settings based on the selected vector store provider.

eshurakov committed:
Merge branch 'main' into pr/2587

eshurakov committed:
Update pnpm-lock.yaml

eshurakov committed:
Normalize directoryPrefix to relative path

eshurakov committed:
Improve SQL string escaping and query filtering

eshurakov committed:
Merge branch 'main' into feature-vector-store-lancedb

eshurakov committed:
Revert whitespace changes to docs

eshurakov committed:
Add kilocode_change

eshurakov committed:
Changeset

eshurakov reviewed (approved):
Thank you for the PR, it looks good to me
